src/fromager/bootstrapper.py

```diff
@@ -316,11 +303,81 @@ def _handle_prebuilt_sources(
+        try:
+            wheel_url, _ = resolver.resolve(
+                ctx=self.ctx,
+                req=Requirement(f"{req.name}=={resolved_version}"),
+                sdist_server_url=self.cache_wheel_server_url,
+                include_sdists=False,
+                include_wheels=True,
+            )
+            cached_wheel = wheels.download_wheel(
+                req=req, wheel_url=wheel_url, output_directory=self.ctx.wheels_downloads
+            )
+            server.update_wheel_mirror(self.ctx)
+
+            pbi = self.ctx.package_build_info(req)
+            expected_build_tag = pbi.build_tag(resolved_version)
+            dist_name, dist_version, build_tag, _ = wheels.extract_info_from_wheel_file(
+                req, cached_wheel
+            )
+            if expected_build_tag and expected_build_tag != build_tag:
+                logger.info(
+                    f"{req.name}: found wheel for {resolved_version} in cache but build tag does not match. Got {build_tag} but expected {expected_build_tag}"
+                )
+                return None, None
+
+            logger.info(f"{req.name}: found built wheel on cache server")
+            unpack_dir = self._create_unpack_dir(req, resolved_version)
+            # unpack the wheel
+            external_commands.run(
+                ["wheel", "unpack", str(cached_wheel), "--dest", unpack_dir],
+                cwd=unpack_dir,
+            )
+
+            dist_filename = f"{dist_name}-{dist_version}"
+            metadata_dir = unpack_dir / dist_filename / f"{dist_filename}.dist-info"
+
+            # prepare files cached by fromager by placing them in the parent directory of the return path
+            try:
+                shutil.copy(
+                    metadata_dir
+                    / f"{wheels.FROMAGER_BUILD_REQ_PREFIX}-{dependencies.BUILD_SYSTEM_REQ_FILE_NAME}",
+                    unpack_dir / dependencies.BUILD_SYSTEM_REQ_FILE_NAME,
+                )
+                shutil.copy(
+                    metadata_dir
+                    / f"{wheels.FROMAGER_BUILD_REQ_PREFIX}-{dependencies.BUILD_BACKEND_REQ_FILE_NAME}",
+                    unpack_dir / dependencies.BUILD_BACKEND_REQ_FILE_NAME,
+                )
+                shutil.copy(
+                    metadata_dir
+                    / f"{wheels.FROMAGER_BUILD_REQ_PREFIX}-{dependencies.BUILD_SDIST_REQ_FILE_NAME}",
+                    unpack_dir / dependencies.BUILD_SDIST_REQ_FILE_NAME,
+                )
+                logger.info(f"{req.name}: extracted build requirements from wheel")
+                return unpack_dir / dist_filename, cached_wheel
+            except Exception:
+                # implies that the wheel server hosted non-fromager built wheels
+                logger.info(
+                    f"{req.name}: could not extract build requirements from wheel"
+                )
+                return None, cached_wheel
```

Comment:
If we can't extract all of the requirements info, should we delete whatever we could extract so that we recompute it? Or does this return value trigger that behavior?

Reply:
The return value triggers that behavior. This handles the case where the cache wheel server url is hosting non-fromager built wheels. This will allow the user to still skip building the wheel itself but will trigger a recomputation of the build requirements

Reply:
ah actually you are right we do need to do some cleanup